File: C++_from_R_src/Fibonacci.c

#include <R.h>
// typically n would not be a pointer
void Fibonacci(int* n, int* f)
  {
  if(*n < 2)
    *n = -1;
  else
    {
    f[0] = 0;
    f[1] = 1;
    for(int i = 2; i<*n; i++)
      f[i] = f[i-1]+f[i-2];
    }
  return;
  }

File: C++_from_R_src/Fibonacci.R

setwd('/Users/pawankarki/Desktop/sta705/C++_from_R_src')
dyn.load("Fibonacci.so")

n      <- 10
# create memory for the result
f      <- vector(mode = "integer", length = n);
# f    <- rep(0, n)   # this would also work
result <- .C("Fibonacci", as.integer(n), as.integer(f))
# result contains a list with an element for
# each argument
print( result[[2]] )

